package/com.unity.formats.usd/Tests/USD.NET.Unity/UsdPreviewSurfaceTests.cs

```diff
@@ -8,76 +8,142 @@ namespace USD.NET.Unity.Tests
-            Assert.AreEqual(shader.diffuseColor.connectedPath, param.connectedPath);
+        void CheckShaderParams()
+        {
+            var param = m_USDReadShader.GetInputParameters().First();
+            Assert.AreEqual(m_originalShader.diffuseColor.connectedPath, param.connectedPath);
             Assert.AreEqual("diffuseColor", param.usdName);
-            Assert.AreEqual(shader.diffuseColor.defaultValue, param.value);
+            Assert.AreEqual(m_originalShader.diffuseColor.defaultValue, param.value);
             Assert.AreEqual("_DiffuseColor", param.unityName);
+        }
 
-            Assert.AreEqual(material.surface.defaultValue, material2.surface.defaultValue);
-            Assert.AreEqual(material.surface.connectedPath, material2.surface.connectedPath);
-            Assert.AreEqual(shader.diffuseColor.defaultValue, shader2.diffuseColor.defaultValue);
-            Assert.AreEqual(shader.diffuseColor.connectedPath, shader2.diffuseColor.connectedPath);
-            Assert.AreEqual(shader.id, shader2.id);
-            Assert.AreEqual(texture.file.defaultValue, texture2.file.defaultValue);
-            Assert.AreEqual(primvarReader.varname.defaultValue, primvarReader.varname.defaultValue);
+        void CheckMaterialValues()
+        {
+            Assert.AreEqual(m_originalMaterial.surface.defaultValue, m_USDReadMaterial.surface.defaultValue);
+            Assert.AreEqual(m_originalMaterial.surface.connectedPath, m_USDReadMaterial.surface.connectedPath);
+            Assert.AreEqual(m_originalShader.diffuseColor.defaultValue, m_USDReadShader.diffuseColor.defaultValue);
+            Assert.AreEqual(m_originalShader.diffuseColor.connectedPath, m_USDReadShader.diffuseColor.connectedPath);
+            Assert.AreEqual(m_originalShader.id, m_USDReadShader.id);
+            Assert.AreEqual(m_originalTexture.file.defaultValue, m_USDReadTexture.file.defaultValue);
         }
+
+        // Use the 'export' version of the PrimvarReaderSample, because this uses a TfToken typed varname to match the USD 20.08 specification instead of
+        // the 21.11+ spec of string.
+        [Test]
+        public void WritingAndReadingMaterialToUSDScene_ValuesAreCorrect()
+        {
+            PrimvarReaderExportSample<Vector2> primvarReaderTfToken = new PrimvarReaderExportSample<Vector2> { varname = { defaultValue = new TfToken("st") } };
+
+            m_USDScene.Write(k_primvarReaderPath, primvarReaderTfToken);
+
+            MaterialSample.Bind(m_USDScene, k_cubePath, k_materialPath);
+
+            PrimvarReaderImportSample<Vector2> USDReadPrimvarReader = new PrimvarReaderImportSample<Vector2>();
+
+            ReadPrimsFromScene();
+            m_USDScene.Read(k_primvarReaderPath, USDReadPrimvarReader);
+
+            CheckShaderParams();
+            CheckMaterialValues();
+
+            Assert.AreEqual(primvarReaderTfToken.varname.defaultValue, primvarReaderTfToken.varname.defaultValue);
```

Comment:
Should one of these parameters refer to USDReadPrimvarReader?

Reply:
Yes :) Thanks!